core/src/main/java/io/undertow/attribute/ResponseSizeAttribute.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.undertow.attribute;
+
+import io.undertow.server.HttpServerExchange;
+import io.undertow.util.StatusCodes;
+
+/**
+ * Size of response in bytes, including headers
+ *
+ * @author Marek Jusko
+ */
+
+public class ResponseSizeAttribute implements ExchangeAttribute{
+
+    public static final String RESPONSE_SIZE_SHORT = "%O";
+    public static final String RESPONSE_SIZE = "%{RESPONSE_SIZE}";
+    public static final ExchangeAttribute INSTANCE = new ResponseSizeAttribute();
+
+    @Override
+    public String readAttribute(HttpServerExchange exchange) {
```

Comment:
What if HTTP 2 protocol is in use? The calculations here expect HTTP 1.1 is being used.

Reply:
good point, but AFAIR we cant account for, for instance, header  compression? Also, even for H1 with compression this will be off?
@xjusko ^^ ?